### ISSUE TITLE
Messages sent after device sleep may stuck in "sending" state in community

### ISSUE THREAD

**Issue by churik** [status-im/status-react#18284]
# Bug Report

## Problem 
I believe the issue exists regardless of the fleet, but in the current release on `shards.test` version it is more noticeable. 
Messages are completely stuck in "sending" state if they are sent after the device is unlocked. Re-login doesn't help, the messages are stuck there forever and they are not being resent. 

#### Expected behavior
Messages are sent 

#### Actual behavior
messages are stuck there forever and they are not being resent even after re-login 

### Reproduction
**Prerequisites:** device A joined to community 

1. Lock the device and wait ~5 mins (app should be on the background)
2. Unlock the device and try to send a couple of messages to community

### Additional Information

- Status version: release 18 
- Operating System: Android, iOS

[comment]: # (Please, add logs/notes if necessary)

**Comment by churik** [comment]
Currently can reproduce consistently with android sending from offline state - messages stuck in "Sending" state forever after getting back online
![IMAGE 2023-12-26 11:48:34](https://github.com/status-im/status-mobile/assets/4557972/68a77b46-b2df-4c8a-9aa1-bd0d97eedca0)


Logs: [Status-debug-logs.zip](https://github.com/status-im/status-mobile/files/13770902/Status-debug-logs.zip)

**Comment by pavloburykh** [comment]
**UPDATE** (22.01.24):

So far I am unable to reproduce this issue. Maybe something has been fixed on go-waku side recently, not sure.

But let's keep this issue open for a while. If we face it again we will post an update here.

**Comment by pavloburykh** [comment]
**UPDATE** (25.01.24): 

I have been able to reproduce the bug on the recovered user. In my case the issue seems nothing to do with sleep mode or offline/online handling. 

The issues have been reproduced both in communities and 1-1 chats. 

Some of the messages have subsequently changed status from Sending to Sent and have been in fact delivered. 
But some of the messages have been stuck in Sending status forever and have not been delivered (or delivered partially, i.e message contained text + 6 images but only text + 3 images have been delivered and 3 images have been lost). 

[Status-debug-logs (97).zip](https://github.com/status-im/status-mobile/files/14052742/Status-debug-logs.97.zip)

![photo_2024-01-25 15 33 14](https://github.com/status-im/status-mobile/assets/97245802/3237d2db-d377-4254-8db3-64f42a64c345)
![photo_2024-01-25 15 33 18](https://github.com/status-im/status-mobile/assets/97245802/a02ad493-d86a-4a65-84a4-4a292f7f1c31)

**Comment by ulisesmac** [comment]
I tried to replicate the issue several times but I couldn't. Has it been solved? I tested on two physical Android devices and an emulator.

Tried following the steps in the description and also as specified in the comment https://github.com/status-im/status-mobile/issues/18284#issuecomment-1869453182 
CC: @churik 

This issue could be closed or maybe we need to specify exact steps to reproducte it :thinking:
